### PR TITLE
MultiSelect: Add status and validationMessage props

### PR DESCRIPTION
**Pull request by mcwinter07**
## Why
Adding missing features

## What
-  Add `status` variants styles and sitckersheet variants to MultiSelectToggle
- Add `validationMessage` and `status` to MultiSelect
  - Connect `validationMessage` to `aria-describedby`
  - Add test coverage for validation states
  - Add sitckersheet variants

### CODE REVIEW

File: packages/components/src/MultiSelect/MultiSelect.spec.tsx
Comment (mcwinter07) [inline packages/components/src/MultiSelect/MultiSelect.spec.tsx:2]: Using screen here since this is the decided way going forward but will hold off on refactoring the others to use this to a later time

File: packages/components/src/MultiSelect/MultiSelect.tsx
Comment (mcwinter07) [inline packages/components/src/MultiSelect/MultiSelect.tsx:26]: Noting that we are only really using `error` and `caution` from this. Is it worth just typing those two or leaving as is? Neither our `default` or `success` state impact the UI?

File: packages/components/src/MultiSelect/MultiSelect.tsx
Comment (mcwinter07) [inline packages/components/src/MultiSelect/MultiSelect.tsx:87]: This will probably conflict with Dug's PR so may need to loop back here if that gets merged first

File: packages/components/src/MultiSelect/MultiSelect.tsx
Comment (HeartSquared) [reply packages/components/src/MultiSelect/MultiSelect.tsx:26]: You could use `Pick`? Probably better to only include the ones necessary, otherwise it will look like a bug 😬

File: packages/components/src/MultiSelect/MultiSelect.spec.tsx
Comment (HeartSquared) [inline packages/components/src/MultiSelect/MultiSelect.spec.tsx:296]: Definitely worth adding a test (or amending this one) to also showcase the order relative to the field description when Doug's PR is merged in.

File: packages/components/src/MultiSelect/MultiSelect.tsx
Comment (HeartSquared) [inline packages/components/src/MultiSelect/MultiSelect.tsx:43]: You'll be needing to combine with the description, so they need different ids :)
```suggestion
  const validationId = `${id}--validation-message`
```

File: packages/components/src/MultiSelect/_docs/MultiSelect.stickersheet.stories.tsx
Comment (HeartSquared) [inline packages/components/src/MultiSelect/_docs/MultiSelect.stickersheet.stories.tsx:137]: This shouldn't be needed?

File: packages/components/src/MultiSelect/subcomponents/MultiSelectToggle/MultiSelectToggle.tsx
Comment (HeartSquared) [inline packages/components/src/MultiSelect/subcomponents/MultiSelectToggle/MultiSelectToggle.tsx:49]: The array isn't needed :)
```suggestion
        className={classnames(styles.multiSelectToggle, classNameOverride,
          status && styles[status],
        )}
```

File: packages/components/src/MultiSelect/MultiSelect.spec.tsx
Comment (mcwinter07) [reply packages/components/src/MultiSelect/MultiSelect.spec.tsx:294]: True, I think I'll change this to "has a validation message" since I broke out the accessibility testing into the other test

File: packages/components/src/MultiSelect/MultiSelect.tsx
Comment (mcwinter07) [reply packages/components/src/MultiSelect/MultiSelect.tsx:26]: Ah I think this has to be `Extract` since `FieldMessageStatus` is a union of string literals

File: packages/components/src/MultiSelect/subcomponents/MultiSelectToggle/_docs/MultiSelectToggle.stickersheet.stories.tsx
Comment (mcwinter07) [reply packages/components/src/MultiSelect/subcomponents/MultiSelectToggle/_docs/MultiSelectToggle.stickersheet.stories.tsx:128]: You could argue that the hover and focus change is tested in the previous story since our border don't change for validation states. Though I do wonder if there's value in it to ensure that there is a11y contrast in the story.

Comment (mcwinter07) [reply packages/components/src/MultiSelect/subcomponents/MultiSelectToggle/_docs/MultiSelectToggle.stickersheet.stories.tsx:128]: Maybe I'll remove them and we can add the hover and focus states back into the Toggle if ever add any additional styles (like a darkened border)

Comment (HeartSquared) [reply packages/components/src/MultiSelect/subcomponents/MultiSelectToggle/_docs/MultiSelectToggle.stickersheet.stories.tsx]: Yeah I think if they don't have anything specific, it should be safe to not have them :)

File: packages/components/src/MultiSelect/MultiSelect.spec.tsx
Comment (mcwinter07) [reply packages/components/src/MultiSelect/MultiSelect.spec.tsx]: Added in a new test to testing the concatenation specifically in the accessible description 👍

File: packages/components/src/MultiSelect/MultiSelect.tsx
Comment (gyfchong) [reply packages/components/src/MultiSelect/MultiSelect.tsx]: Yes please!

Comment (gyfchong) [reply packages/components/src/MultiSelect/MultiSelect.tsx]: (Apologies @dougmacknz this wasn't a very well documented pattern)

File: packages/components/src/MultiSelect/types.ts
Comment (gyfchong) [inline packages/components/src/MultiSelect/types.ts:12]: Seems like something we need to make more widely accessible to the other fields

File: packages/components/src/MultiSelect/subcomponents/MultiSelectToggle/MultiSelectToggle.tsx
Comment (gyfchong) [inline packages/components/src/MultiSelect/subcomponents/MultiSelectToggle/MultiSelectToggle.tsx:4]: Should this be here?

File: packages/components/src/MultiSelect/types.ts
Comment (mcwinter07) [reply packages/components/src/MultiSelect/types.ts:12]: You mean like a shared types file in the root of the src similar to the style utils?

Comment (gyfchong) [reply packages/components/src/MultiSelect/types.ts:12]: Could it belong to a component?

Comment (mcwinter07) [reply packages/components/src/MultiSelect/types.ts:12]: We could export the type from FieldMessage since these props are consumed in that component - assuming that we only want validation messages to have `error` and `caution` states.

Comment (gyfchong) [reply packages/components/src/MultiSelect/types.ts:12]: We can always extend later, but its your call whether it makes sense to do it now 😄

File: packages/components/src/MultiSelect/subcomponents/MultiSelectToggle/MultiSelectToggle.tsx
Comment (mcwinter07) [reply packages/components/src/MultiSelect/subcomponents/MultiSelectToggle/MultiSelectToggle.tsx:4]: Not anymore, just a hangover of a rebase, will 🪓

File: packages/components/src/MultiSelect/types.ts
Comment (mcwinter07) [reply packages/components/src/MultiSelect/types.ts:12]: My main concern was the existing one in the Filter Date Picker, since it has all states accounted for, but looking at the story its only ever used for a error state. As you say we can always extend on it and it doesn't hurt to be more specific in what we're expecting

Comment (mcwinter07) [reply packages/components/src/MultiSelect/types.ts]: Hmmm gave it a shot and realized I'd have to make a fair few changes to the imports in the Filter DP. I think I'll revert that back and create a ticket following this one to move that type to the FieldMessage component once this is merged. I'd rather untangle this release and get a specific optimization out as its own thing :)

**Review by gyfchong — APPROVED** [review]

**Pull request merged**
